lib/gradient/tokens.ex

```diff
@@ -0,0 +1,198 @@
+defmodule Gradient.Tokens do
+  @moduledoc """
+  Group of functions helping with manage tokens.
+  """
+  alias Gradient.Types, as: T
+
+  @doc """
+  Drop tokens to the first conditional occurance. Returns type of the encountered
+  conditional and following tokens.
+  """
+  @spec get_conditional(T.tokens(), integer(), T.options()) ::
+          {:case, T.tokens()}
```

Comment:
This return value has 6 variants, so it's quite big. Might be worth extracting to its own top-level type for readability.

Reply:
Done in one of the next commits